### PR TITLE
Add Distributed Workload Metrics Ui Tests

### PR DESCRIPTION
**Pull request by Srihari1192**
closes [RHOAIENG-4837](https://issues.redhat.com/browse/RHOAIENG-4837)

- Added Distributed workload Metrics basic tests to check with **admin and non admin users**
-  Added common methods and required xpath variables in **WorkloadMetricsUi.resource** page

### CODE REVIEW

File: ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot
Comment (Srihari1192) [reply ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:78]: This can be removed later once disableDistributedWorkloads enabled by default.

File: ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot
Comment (Srihari1192) [reply ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:89]: This can be removed later once disableDistributedWorkloads enabled by default.

File: ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot
Comment (FedeAlonso) [inline ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:22]: Could you add some tag in the tests? RHOAIENG-4837 could be the good one (like it's done [here](https://github.com/red-hat-data-services/ods-ci/blob/4fee8773623707d9f1c6fecf291fd6663d960b1e/ods_ci/tests/Tests/400__ods_dashboard/416__ods_dashboard_settings_accelerator_profiles.robot#L34))

**Review by FedeAlonso — CHANGES_REQUESTED** [review]

File: ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot
Comment (sutaakar) [inline ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:69]: Shouldn't this be part of `Project Suite Teardown` rather than being part of the test itself?

File: ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot
Comment (mturley) [inline ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:114]: This text may change before release, will let you know. We don't have filters implemented yet in the table so we may be replacing the whole table with an empty state message when there are no workloads.

File: ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot
Comment (Srihari1192) [reply ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:69]: We can delete all the created Projects in `Project Suite Teardown `using method `Delete Data Science Projects From CLI` but thinking  if we ran tests independently this project may not be available and test fails

File: ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot
Comment (Srihari1192) [reply ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:22]: sure will update the Tags , Thanks

File: ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot
Comment (sutaakar) [reply ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:69]: ok

File: ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot
Comment (sutaakar) [inline ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:53]: I think that this may be executed as part of Tier 1, no need to run as sanity test.
This test isn't a core basic test.

Comment (Srihari1192) [reply ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:53]: Yes will update the Tags

**Review by sutaakar — APPROVED** [review]

File: ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot
Comment (sutaakar) [inline ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:76]: Does it mean that whole test suite is skipped or just setup?

Comment (Srihari1192) [reply ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:76]: Whole test suite is skipped.. I added this because DW Ui is disabled if kueue is `Removed`

File: ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot
Comment (Srihari1192) [reply ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:80]: yes ideally it update existing dashboard pods and already in running state so adding condition to pod state also not the right approach i think.. 
Default enabling DW UI will be available with this  [PR](https://github.com/opendatahub-io/odh-dashboard/pull/2685) merge soon

File: ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot
Comment (mturley) [reply ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:114]: FYI @Srihari1192 now that https://github.com/opendatahub-io/odh-dashboard/pull/2686 is merged, this text is now simply "No distributed workloads"

File: ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot
Comment (sutaakar) [reply ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:80]: So, once the linked PR is merged then there won't be a need for any sleep step, right?

Comment (Srihari1192) [reply ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:80]: Yes we can remove this sleep once its merged

Comment (sutaakar) [reply ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:80]: Looks like that PR should be hopefully merged soon.
Do you want to wait for it to merge first or rather merge this PR now and remove sleep as follow up?

Comment (Srihari1192) [reply ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:80]: Yes we can merge this PR for now .. i will update later once both changes [pr1](https://github.com/opendatahub-io/odh-dashboard/pull/2685) and [pr2](https://github.com/opendatahub-io/odh-dashboard/pull/2686) available in rhoai builds

**Review by sutaakar — APPROVED** [review]

File: ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot
Comment (mturley) [reply ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:80]: https://github.com/opendatahub-io/odh-dashboard/pull/2685 is now merged, builds will have the feature flag on by default going forward

**Comment by Srihari1192** [comment]
Added Kueue resources setup from the [PR](https://github.com/project-codeflare/codeflare-sdk/pull/505) Thanks @ChughShilpa for implementing this.
Updated the PR according to the new UI changes https://github.com/opendatahub-io/odh-dashboard/pull/2686 and https://github.com/opendatahub-io/odh-dashboard/pull/2685

File: ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot
Comment (sutaakar) [inline ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:95]: Can you make sure that resources are also deleted after test execution?

Comment (Srihari1192) [reply ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot:95]: sure

Comment (Srihari1192) [reply ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot]: tested in PR check CI pipeline `rhods-ci-pr-test/2703`

File: ods_ci/tests/Resources/Page/DistributedWorkloads/kueue_resources_setup.sh
Comment (sutaakar) [inline ods_ci/tests/Resources/Page/DistributedWorkloads/kueue_resources_setup.sh:22]:
```suggestion
        - name: $flavor
```

**Comment by sutaakar** [comment]
One fix above, otherwise looks good

File: ods_ci/tests/Resources/Page/DistributedWorkloads/kueue_resources_setup.sh
Comment (Srihari1192) [reply ods_ci/tests/Resources/Page/DistributedWorkloads/kueue_resources_setup.sh]: good catch , Thanks

**Review by sutaakar — APPROVED** [review]

**Review by jiripetrlik — APPROVED** [review]

**Review by ChughShilpa — APPROVED** [review]
Tested locally  and passed all tests

File: ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot
Comment (Srihari1192) [reply ods_ci/tests/Tests/650__distributed_workloads/test-distributed-workloads-metrics-ui.robot]: @FedeAlonso Can you review this Updated Tags

**Review by FedeAlonso — APPROVED** [review]

**Pull request merged**
